src/Widget/Form.php

```diff
@@ -61,51 +73,195 @@ public function begin(): ?string
+     */
+    public function autocomplete(string $value = 'on'): self
+    {
+        if ($value !== 'on' && $value !== 'off') {
+            throw new InvalidArgumentException('The value must be `on`,` off`.');
+        }
+
+        $new = clone $this;
+        $new->attributes['autocomplete'] = $value;
+        return $new;
+    }
+
+    /**
+     * The csrf-token content attribute is a space-separated list of tokens that are known to be safe to use for.
+     *
+     * @param string $value the csrf-token attribute value.
+     *
+     * @return static
+     */
+    public function csrf(string $value): self
+    {
+        $new = clone $this;
+        $new->attributes['_csrf'] = $value;
+        return $new;
+    }
+
+    /**
+     * The formenctype content attribute specifies the content type of the form submission.
+     *
+     * @param string $value the formenctype attribute value.
+     *
+     * @return static
+     *
+     * @link https://www.w3.org/TR/html52/sec-forms.html#element-attrdef-form-enctype
+     */
+    public function enctype(string $value): self
+    {
+        $new = clone $this;
+        $new->id = $value;
+        return $new;
+    }
+
+    /**
+     * The id content attribute is a unique identifier for the element.
+     *
+     * @param string $value the id attribute value.
+     *
+     * @return static
+     */
+    public function id(string $value): self
+    {
+        $new = clone $this;
+        $new->id = $value;
+        return $new;
+    }
+
+    /**
+     * The method content attribute specifies how the form-data should be submitted.
+     *
+     * @param string $value the method attribute value.
+     *
+     * @return static
+     *
+     * @link https://www.w3.org/TR/html52/sec-forms.html#element-attrdef-form-method
+     */
     public function method(string $value): self
     {
         $new = clone $this;
-        $new->method = $value;
+        $new->method = strtoupper($value);
         return $new;
     }
 
-    public function options(array $value = []): self
+    /**
+     * The novalidate and formnovalidate content attributes are boolean attributes. If present, they indicate that the
+     * form is not to be validated during submission.
+     *
+     * @return static
+     *
+     * @link https://www.w3.org/TR/html52/sec-forms.html#element-attrdef-form-novalidate
+     */
+    public function noValidateHtml(): self
     {
         $new = clone $this;
-        $new->options = $value;
+        $new->attributes['novalidate'] = true;
         return $new;
     }
+
+    /**
+     * The target and formtarget content attributes, if specified, must have values that are valid browsing context
+     * names or keywords.
+     *
+     * @param string $value the target attribute value, for default its `_blank`.
+     *
+     * @return static
+     *
+     * @link https://www.w3.org/TR/html52/sec-forms.html#element-attrdef-form-target
+     */
+    public function target(string $value = '_blank'): self
```

Comment:
I'd not set a default here because `_blank` isn't really secure.

Reply:
done.